### ISSUE TITLE
Hyperpod cluster creation fails if Docker is installed via LifeCycle scripts

### ISSUE THREAD

**Issue by awsankur** [aws-samples/awsome-distributed-training#96]
Lifecycle scripts do not run successfully `ExecuteBashScript("./utils/install_docker.sh").run()` is uncommented.

The Cloudwatch logs just mention:

```
subprocess.CalledProcessError: Command '['sudo', 'bash', './utils/install_docker.sh']' returned non-zero exit status 100.
```

**Comment by sean-smith** [comment]
I believe the issue here is it can't get the apt-get lock here: https://github.com/aws-samples/awsome-distributed-training/blob/main/1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config/utils/install_docker.sh#L5

**Comment by johnbensnyder** [comment]
I seem to remember this coming up sporadically on larger clusters. Maybe adding a timeout period to retry would help? Something like `sudo apt-get -y -o DPkg::Lock::Timeout=120 update`?

**Comment by verdimrc** [comment]
can we verify if Docker is already pre-installed in Hyperpod? If so, that script can be deprecated or even removed from existence.

DLAMI base has been preinstalled with Docker (+nvidia container runtime) for a long time, and in recent months the framework variant (of DLAMI) also pre-installed with those container runtime.

**Comment by sean-smith** [comment]
@verdimrc it's not

**Comment by verdimrc** [comment]
Closed via #127 (apply suggested fix in this [comment](https://github.com/aws-samples/awsome-distributed-training/issues/96#issuecomment-1890297370)).
